### PR TITLE
Wrapper: Add class for wrapping a native tool

### PR DESCRIPTION
**Pull request by diorcety**

### CODE REVIEW

File: classes/wrapper.oeclass [hunk 1]
```diff
@@ -0,0 +1,55 @@
+## Class for packages having wrapper files.
+##
+## Rewrites the paths in the wrapper files so that it contains valid dirs.
+
+require conf/meta.conf
+
+WRAPPER_FILES ?= "${bindir}/*"
+WRAPPER_GLOB ?= "${WRAPPER_FILES}"
```
Comment (esben) [inline classes/wrapper.oeclass]: Why both WRAPPER_FILES and WRAPPER_GLOB ?

File: classes/wrapper.oeclass [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    wrapper_globs = d.get("WRAPPER_GLOB").lstrip("/").split()
+    os.chdir(pkgd)
+    for pkg in os.listdir("."):
+        os.chdir(os.path.join(pkgd, pkg))
+        wrapper_files = []
+        for wrapper_glob in wrapper_globs:
+            for wrapper_file in glob.glob(wrapper_glob.lstrip("/")):
+                wrapper_dotpath = "%s/.%s"%(os.path.dirname(wrapper_file), os.path.basename(wrapper_file))
+   
+                localpath = oelite.path.which(d.get("FILESPATH_EXISTS"), wrapper_file)
```
Comment (esben) [inline classes/wrapper.oeclass]: What is going on here?  Are you looking back at files in the manifest in a do_split postfunc?

In do_split you should only be concerned with the files in PKGD.

File: classes/wrapper.oeclass [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                if localpath:
+                    print "wrapper file %s -> %s" % (wrapper_file, wrapper_dotpath)
+                    wrapper_files.append("/" + wrapper_file)
+                    os.rename(wrapper_file, wrapper_dotpath)
+                    with open(localpath, "r") as input_file:
```
Comment (esben) [inline classes/wrapper.oeclass]: Why not open the wrapper_file (at this point, renamed to wrapper_dotpath)?

File: classes/wrapper.oeclass [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                        content = input_file.read()
+                    content = re.sub(
+                        re.compile('\$\{realfile\}', re.MULTILINE),
```
Comment (esben) [inline classes/wrapper.oeclass]: Hm...  Could you add some description on when and how to create such wrapper files?

File: classes/wrapper.oeclass (hunk 1 above)
Comment (diorcety) [inline classes/wrapper.oeclass]: brainless copy/paste/sed on binconfig :D sorry

File: classes/wrapper.oeclass (hunk 4 above)
Comment (diorcety) [inline classes/wrapper.oeclass]: Can find an example in my base PR in file ./recipes/bison/bison/bin/bison

    #!/bin/sh
    export BISON_PKGDATADIR=${datadir}/bison
    exec -a ${exec_prefix}${currentfile} ${exec_prefix}${realfile} $@

**Comment by esben** [comment]
@diorcety could you look at my comments on this PR?  I would like to get this merged in time for 5.0.0

**Comment by diorcety** [comment]
@esben Oups sorry, i have done some changing according to your comments but i haven't tested it. I will now and push the modifications :)

File: classes/wrapper.oeclass (hunk 2 above)
Comment (diorcety) [inline classes/wrapper.oeclass]: What do you mean?

Comment (esben) [inline classes/wrapper.oeclass]: Calling oelite.path.which with FILESPATH_EXISTS, means you are looking for files in recipe dir.  This should only be done by fetcher.  All files you need from recipe dir, should be included in SRC_URI, and accessed via fx. SRCDIR in other tasks.

Comment (diorcety) [inline classes/wrapper.oeclass]: Okay i have another main issue with my work. That i put that at the split which is not really correct. But if i put that after install i will be have issue with unpackaged files ...

File: classes/wrapper-stage.oeclass [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+WRAPPER_STAGE_DIRNAMES = "prefix exec_prefix bindir sbindir libdir \
+	includedir libexecdir datarootdir datadir sysconfdir sharedstatedir \
+	localstatedir infodir mandir"
+
+# FIXME: add a method for packages providing wrapper files to specify which
+# dirnames to fixup, so that fx. this installbuilddir could be provided by apr
+# recipe, so that it does not impact metadata, and thus signature of all
+# recipes.
+WRAPPER_STAGE_DIRNAMES += "installbuilddir"
```
Comment (esben) [inline classes/wrapper-stage.oeclass]: What is wrong with just keeping this append in the recipe (after inherit wrapper-stage)?

Comment (diorcety) [inline classes/wrapper-stage.oeclass]: The shame of the copy paste. Shame on me :(

**Comment by esben** [comment]
I can merge it as it is now.

But if possible, take a look at my comment on your FIXME thing.  Would be nice to have removed that before 5.0.0 release.

File: classes/wrapper-stage.oeclass (hunk 5 above)
Comment (esben) [inline classes/wrapper-stage.oeclass]: Hehe.

But you only removed the FIXME comment.  What about the installbuilddir append, couldn't you just remove that and do it in the apr recipe instead?

**Pull request merged**